apollo-federation/src/sources/connect/expand/visitors/selection.rs

```diff
@@ -134,21 +142,26 @@ impl GroupVisitor<JSONSelectionGroup, NamedSelection>
         let (definition, _) = self.type_stack.last().ok_or(FederationError::internal(
             "tried to get fields on a group not yet visited",
         ))?;
-        let field_name = Name::new(field.name())?;
-
-        let field_type_name = definition
-            .field(field_name)
-            .get(self.original_schema.schema())?
-            .ty
-            .inner_named_type();
-
-        let TypeDefinitionPosition::Object(field_type) =
-            self.original_schema.get_type(field_type_name.clone())?
-        else {
-            return Ok(None);
-        };
 
-        Ok(field.next_subselection().cloned().map(|s| (field_type, s)))
+        match field.names().as_slice() {
+            &[field_name] => {
```

Comment:
Could maybe be `field.names().first()`?

Reply:
https://github.com/apollographql/router/pull/6076/commits/560d27d21fd303e4b00db563539e54a171f92ee4